source/renderer/app/stores/HardwareWalletsStore.ts

```diff
@@ -1786,6 +1573,403 @@ export default class HardwareWalletsStore extends Store {
       throw error;
     }
   };
+
+  @action
+  _recognizeSoftwareWalletByExtendedPublicKey = async ({
+    extendedPublicKey,
+  }: RecognizeSoftwareWalletByExtendedPublicKeyArgs): Promise<Wallet | null> => {
+    const deviceId =
+      extendedPublicKey?.deviceId || this.transportDevice.deviceId;
+    logger.debug('[HW-DEBUG] HWStore - EXPORT - deviceID: ', {
+      deviceId,
+    });
+
+    const recognizedStoredWallet = find(
+      this.hardwareWalletsConnectionData,
+      (hardwareWalletData) =>
+        extendedPublicKey.chainCodeHex ===
+          hardwareWalletData.extendedPublicKey.chainCodeHex &&
+        extendedPublicKey.publicKeyHex ===
+          hardwareWalletData.extendedPublicKey.publicKeyHex
+    );
+
+    return recognizedStoredWallet
+      ? this.stores.wallets.getWalletById(recognizedStoredWallet.id)
+      : null;
+  };
+
+  // Delete initiated (pending) device with this path since now is paired to wallet
+  @action
+  _deletePendingDeviceWithGivenPath = async ({
+    path,
+  }: DeletePendingDeviceWithGivenPathArgs) => {
+    const recognizedDevice = find(
+      this.hardwareWalletDevices,
+      // @ts-ignore ts-migrate(2339) FIXME: Property 'path' does not exist on type 'HardwareWa... Remove this comment to see the full error message
+      (device) => device.path === path
+    );
+
+    if (!recognizedDevice) return;
+
+    logger.debug(
+      '[HW-DEBUG] HWStore - __deletePendingDeviceWithGivenPath - UNSET Device with path: ',
+      {
+        recognizedDevice: recognizedDevice.id,
+      }
+    );
+    await this._unsetHardwareWalletDevice({
+      deviceId: recognizedDevice.id,
+    });
+  };
+
+  @action
+  _discardConnectedDeviceAndReInitiateTransaction = ({
+    walletId,
+  }: DiscardConnectedDeviceAndReInitiateTransactionArgs = {}) => {
+    // Keep isTransactionInitiated active & Set new device listener by initiating transaction
+    // Show message to reconnect proper software wallet device pair
+    logger.debug(
+      '[HW-DEBUG] unfinishedWalletTxSigning SET: ',
+      // @ts-ignore ts-migrate(2345) FIXME: Argument of type 'string' is not assignable to par... Remove this comment to see the full error message
+      walletId
+    );
+    runInAction(
+      'HardwareWalletsStore:: set HW device CONNECTING FAILED',
+      () => {
+        this.hwDeviceStatus = HwDeviceStatuses.CONNECTING_FAILED;
+        this.activeDevicePath = null;
+        this.unfinishedWalletTxSigning = walletId;
+        this.isExportKeyAborted = false;
+      }
+    );
+  };
+
+  @action
+  _proceedWithTransactionAfterConnectingDevice = ({
```

Comment:
@szymonmaslowski maybe we should call this `_reinitiate... or _recover...`? What do you think?

Reply:
I would say this is rather a continuation of the transaction process as there is the `_signTransactionTrezor` method called at the end of this method. Am I right?